### PR TITLE
Unconditionally generate shape compute FX graph

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122809

Some useful stuff we can do with it like dump it or generate guards
entirely from it, but for now just measure the compile time overhead
of having it on all the time.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/fx/experimental/sym_node.py [hunk 1]
```diff
@@ -117,9 +117,7 @@ def __init__(
         # Record the FX node of the current node if we are doing translation
```
Comment (bdhirsh) [inline torch/fx/experimental/sym_node.py:117]: comment out of date?

File: test/test_dynamic_shapes.py [hunk 2]
```diff
@@ -151,18 +151,27 @@ def create_symbolic_tensor(name, arg, shape_env, source=None, dynamic_dims=None)
...
         dynamic_dim=DimDynamic.DUCK if duck else DimDynamic.DYNAMIC,
         constraint_dim=None,
     )
-    return cls(SymNode(
-        symbol,
-        shape_env,
-        pytype,
-        hint=val,
-    ))
+    if cls is SymInt:
```
Comment (Chillee) [inline test/test_dynamic_shapes.py:161]: Why does this change?

**Review by Chillee — APPROVED** [review]
With benchmarking this kind of stuff I think microbenchmarks might be more useful.

File: test/test_dynamic_shapes.py (hunk 2 above)
Comment (ezyang) [reply test/test_dynamic_shapes.py:161]: Need to hit the SymInt specific ShapeEnv logic which actually allocates an FX node for it

**Comment by ezyang** [comment]
This is a 10x perf regression on this test:

```
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu005.nha1 ~/local/a/pytorch (71423972)]$ python test/inductor/test_torchinductor_dynamic_shapes.py -k test_fuse_large_params_dynamic_shapes_cpu
W0329 08:16:56.926000 140576214230016 torch/_logging/_internal.py:1013] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
frames [('total', 7), ('ok', 7)]
inline_call [("'skip function graph_break in file /data/users/ezyang/a/pytorch/torch/_dynamo/decorators.py'", 2)]
unimplemented []
graph_break [("'skip function graph_break in file /data/users/ezyang/a/pytorch/torch/_dynamo/decorators.py'", 2)]
stats [('calls_captured', 3492), ('unique_graphs', 1)]
inductor [('fxgraph_cache_miss', 1)]
aot_autograd [('total', 1), ('ok', 1)]
.
----------------------------------------------------------------------
Ran 1 test in 260.054s

OK
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu005.nha1 ~/local/a/pytorch (713e4e98)]$ python test/inductor/test_torchinductor_dynamic_shapes.py -k test_fuse_large_params_dynamic_shapes_cpu
W0329 08:23:06.059000 140223581049856 torch/_logging/_internal.py:1013] [0/0] Profiler function <class 'torch.autograd.profiler.record_function'> will be ignored
frames [('total', 7), ('ok', 7)]
inline_call [("'skip function graph_break in file /data/users/ezyang/a/pytorch/torch/_dynamo/decorators.py'", 2)]
unimplemented []
graph_break [("'skip function graph_break in file /data/users/ezyang/a/pytorch/torch/_dynamo/decorators.py'", 2)]
stats [('calls_captured', 3492), ('unique_graphs', 1)]
inductor [('fxgraph_cache_miss', 1)]
aot_autograd [('total', 1), ('ok', 1)]
.
----------------------------------------------------------------------
Ran 1 test in 2856.462s

OK
```

**Pull request closed without merging**